### PR TITLE
chore: updated dependencies

### PR DESCRIPTION
**Pull request by Sayan751**
# Pull Request

## 📖 Description

Apart from the rollup-dependencies, all the major dependencies are updated. The rollup v3 is causing some issues which can be dealt later. This also adds a deprecation notice to the `@aurelia/store-v1`'s readme.

### CODE REVIEW

File: benchmarks/local/rollup.config.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { nodeResolve } from '@rollup/plugin-node-resolve';
-import { terser } from 'rollup-plugin-terser';
+import terser from '@rollup/plugin-terser';
```
Comment (bigopon) [inline benchmarks/local/rollup.config.js:2]: 👍

File: packages-tooling/plugin-conventions/src/preprocess-resource.ts [hunk 2]
```diff
@@ -227,22 +244,26 @@ function ensureTokenStart(start: number, code: string) {
...
 
-function findDecoratedResourceType(node: ts.Node): IFoundDecorator | void {
-  if (!node.decorators) return;
-  for (const d of node.decorators) {
-    if (!ts.isCallExpression(d.expression)) return;
+function findDecoratedResourceType(node: Node): IFoundDecorator | void {
+  if (!canHaveDecorators(node)) return;
+  const decorators = getDecorators(node);
+  if(decorators === void 0) return;
+  for (const d of decorators) {
```
Comment (bigopon) [inline packages-tooling/plugin-conventions/src/preprocess-resource.ts:263]: looks good 👍

File: packages/__tests__/3-runtime-html/computed-observer.spec.ts [hunk 3]
```diff
@@ -387,7 +387,6 @@ describe('3-runtime-html/computed-observer.spec.ts', function () {
     ({ only, title, template, ViewModel, assertFn }: IComputedObserverTestCase) => {
       // eslint-disable-next-line mocha/no-exclusive-tests
       const $it = (title_: string, fn: Mocha.Func) => only ? it.only(title_, fn) : it(title_, fn);
-      // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (bigopon) [inline packages/__tests__/3-runtime-html/computed-observer.spec.ts:390]: 👍

File: packages/__tests__/router-lite/external.spec.ts [hunk 4]
```diff
@@ -57,7 +57,7 @@ describe('router/external.spec.ts', function () {
 
       assert.strictEqual(host.textContent, '');
       internalLink.click();
-      await router['currentTr'];
+      await router['currentTr'].promise;
```
Comment (bigopon) [inline packages/__tests__/router-lite/external.spec.ts:60]: how did it work before 😮

File: packages/i18n/src/nf/number-format-binding-behavior.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
-import { type BindingBehaviorInstance } from '@aurelia/runtime';
+import type { BindingBehaviorInstance, Scope } from '@aurelia/runtime';
```
Comment (bigopon) [inline packages/i18n/src/nf/number-format-binding-behavior.ts:1]: doing `{ type ... }` eliminates the import from suggestion list, while doing `type { ... }` does not. Sometimes the former is better than the later.

File: packages/kernel/src/di.ts [hunk 6]
```diff
@@ -71,7 +71,7 @@ export interface IContainer extends IServiceLocator, IDisposable {
   registerTransformer<K extends Key, T = K>(key: K, transformer: Transformer<T>): boolean;
   getResolver<K extends Key, T = K>(key: K | Key, autoRegister?: boolean): IResolver<T> | null;
   registerFactory<T extends Constructable>(key: T, factory: IFactory<T>): void;
-  invoke<T, TDeps extends unknown[] = unknown[]>(key: Constructable<T>, dynamicDependencies?: TDeps): T;
+  invoke<T extends {}, TDeps extends unknown[] = unknown[]>(key: Constructable<T>, dynamicDependencies?: TDeps): T;
```
Comment (bigopon) [inline packages/kernel/src/di.ts:74]: 👍

File: packages/rollup-utils.js [hunk 7]
```diff
@@ -317,11 +317,11 @@ function generateNativeModulePlugin(cwd, fileName, enabled) {
...
    */
   async function generateNativeImport(cwd, fileName) {
-    const code = await fs.readFile(path.resolve(cwd, `dist/esm/${fileName}`), { encoding: 'utf-8' });
+    const code = fs.readFileSync(path.resolve(cwd, `dist/esm/${fileName}`), { encoding: 'utf-8' });
     const regex = /from\s+(['"])@aurelia\/([-a-z]+)['"];/g;
     const transformed = code.replace(regex, `from $1../$2/dist/native-modules/${fileName}$1;`).replace(`//# sourceMappingURL=${fileName}.map`, '');
-    await fs.ensureDir(path.resolve(cwd, 'dist/native-modules'));
-    await fs.writeFile(path.resolve(cwd, `dist/native-modules/${fileName}`), transformed, { encoding: 'utf-8' });
+    fs.mkdirSync(path.resolve(cwd, 'dist/native-modules'), { recursive: true });
+    fs.writeFileSync(path.resolve(cwd, `dist/native-modules/${fileName}`), transformed, { encoding: 'utf-8' });
```
Comment (bigopon) [inline packages/rollup-utils.js:324]: 👍

File: packages/store-v1/src/history.ts [hunk 8]
```diff
@@ -59,18 +59,19 @@ export function nextStateHistory<T>(presentStateHistory: StateHistory<T>, nextPr
...
 
-    if (state.future.length > limit) {
-      state.future = state.future.slice(0, limit);
+    if ((state as StateHistory<T>).future.length > limit) {
+      (state as StateHistory<T>).future = (state as StateHistory<T>).future.slice(0, limit);
     }
   }
 
-  return state;
+  return state as T;
```
Comment (bigopon) [inline packages/store-v1/src/history.ts:74]: 👍 
@zewa666

**Review by bigopon — APPROVED** [review]
Nice work, quite a big of tedious change.

File: packages/store-v1/src/history.ts (hunk 8 above)
Comment (zewa666) [reply packages/store-v1/src/history.ts:74]: looks ok although soon no longer important due to deprecation

File: packages/i18n/src/nf/number-format-binding-behavior.ts (hunk 5 above)
Comment (Sayan751) [reply packages/i18n/src/nf/number-format-binding-behavior.ts:1]: Something also to keep in mind when we move to TS 5: https://devblogs.microsoft.com/typescript/announcing-typescript-5-0-beta/#verbatimmodulesyntax

**Comment by Sayan751** [comment]
@bigopon @zewa666 Thanks for the review. I am merging this.

@3cp In case you spot any issue later, please let me know.

**Pull request merged**